src/debugger/gdb-stub.c

```diff
@@ -210,6 +211,62 @@ static void _step(struct GDBStub* stub, const char* message) {
 	UNUSED(message);
 }
 
+static void _writeMemoryBinary(struct GDBStub* stub, const char* message) {
+	const char* readAddress = message;
+	unsigned i = 0;
+	uint32_t address = _readHex(readAddress, &i);
+	readAddress += i + 1;
+
+	i = 0;
+	uint32_t size = _readHex(readAddress, &i);
+	readAddress += i + 1;
+
+	if (size > 512) {
+		_error(stub, GDB_BAD_ARGUMENTS);
+		return;
+	}
+
+	struct ARMCore* cpu = stub->d.core->cpu;
+	for (i = 0; i < size; i++) {
+		// Parse escape char
+		uint8_t byte;
+		if ((byte = *readAddress++) == 0x7D) {
+			byte = *readAddress++ ^ 0x20;
```

Comment:
The size specified as a parameter in the packet is the size in addressable memory units, not the amount of bytes transmitted in the packet. Therefore it's not really possible to check when the packet is over without first reading it. It shouldn't overrun the buffer though, due to the earlier check for `size > 512`. 
Should I make the buffer overrun check more explicit by comparing `size` to `GDB_STUB_MAX_LINE`?